bigframes/bigquery/_operations/sql.py

```diff
@@ -72,16 +73,16 @@ def sql_scalar(
     # Another benefit of this is that if there is a syntax error in the SQL
     # template, then this will fail with an error earlier in the process,
     # aiding users in debugging.
-    base_series = columns[0]
-    literals = [
-        bigframes.dtypes.bigframes_dtype_to_literal(column.dtype) for column in columns
+    literals_sql = [
+        sqlglot_ir._literal(None, column.dtype).sql(dialect="bigquery")
```

Comment:
This might result in some false positives where the dry run succeeds, but there's a type mismatch in the user's SQL. This is because None will always map to NULL in the generated SQL.

Could we use `bigframes_dtype_to_literal` to come up with placeholder values? Or do you think it's not worth maintaining that logic?